spock-core/src/main/java/org/spockframework/runtime/ExpressionInfoValueRenderer.java

```diff
@@ -130,7 +130,7 @@ private String tryReduceStringSizes(String str1, String str2) {
     end1++;
     end2++;
 
-    if ((end1-commonStart) * (end2-commonStart) > MAX_EDIT_DISTANCE_MEMORY) {
+    if (((long) end1-commonStart) * (end2-commonStart) > MAX_EDIT_DISTANCE_MEMORY) {
```

Comment:
shouldn't you need to cast both factors?

Reply:
Yes, the product will be long, but the second factor will only be cast after the subtraction, so the subtraction itself will still remain int-based.

Reply:
I'm with you - this is partly me trying to make the smallest change possible, and also thinking from looking at the code above that since `end1`/`end2`/`end3` should all be positive `int`s, the subtraction shouldn't be a source of overflowing.

I can however look at changing the types of those variables to `long` - do agree it would give us a bit more confidence in the logic. Can do that tonight.

Reply:
Now that I think more about it, you are right: it's not about the subtraction because that cannot overflow anyway. If you only need to widen the multiplication, the current implementation is sufficient.

But maybe you should also mind the similar multiplication happening four lines below, that adds some context to the diff.

Reply:
Good shout, nice spot.

At the moment don't believe this would be an issue due to the value of `MAX_EDIT_DISTANCE_MEMORY` being relatively low, and the original `if` statement we were talking about filtering out any of the high values. Believe it's the reason why I'm currently struggling to write a test showing the wrong behaviour.

However, do think it's still useful to "fix" preemptively as is a little bit of a ticking time-bomb if `MAX_EDIT_DISTANCE_MEMORY` or other depending code is changed in the future. Plus do personally prefer being safe when multiplying `int`s rather than relying on logic further up the code.

Will check in the change now.